Make shr-models a peer dependency

cmoesel opened the pull request:
Due to the nature of the SHR tools, the entire chain must be guaranteed to be using the same instance of the shr-models library.  This has caused issues in the past and required every component of the chain to be updated and re-published whenever shr-models changed in any way.  A better approach to this is to declare shr-models as a peer dependency and rely on the end-user application to bring in the single instance of shr-models.

cmoesel committed:
Make shr-models a peer dependency

Due to the nature of the SHR tools, the entire chain must be guaranteed to be using the same instance of the shr-models library.  This has caused issues in the past and required every component of the chain to be updated and re-published whenever shr-models changed in any way.  A better approach to this is to declare shr-models as a peer dependency and rely on the end-user application to bring in the single instance of shr-models.

jgibson commented on package.json:
```diff
@@ -14,12 +14,14 @@
     "mocha": "^3.2.0"
+  },
+  "peerDependencies": {
+    "shr-models": "^5.2.0"
```
I know that peerDeps are supposed to be [loose](https://nodejs.org/en/blog/npm/peer-dependencies/#using-peer-dependencies), however this standardhealth/shr-models#15 bugfix directly affects at least one of the tests. Could we do `^5.2.1` instead? Or if we really want to drink the semver koolaid we could do `>= 5.2.1 < 6`.

cmoesel replied:
I didn't realize that v5.2.1 affected one of the tests.  Yeah, in that case we can do `^5.2.1` (which, according to my understanding, is exactly equivalent to `>= 5.2.1 < 6`).

jgibson commented on package.json:
```diff
@@ -14,12 +14,14 @@
   "devDependencies": {
     "chai": "^3.5.0",
     "eslint": "^3.6.1",
     "mocha": "^3.2.0"
```
shr-test-helpers doesn't actually appear to use mocha, I think that we can drop the dependency entirely.

cmoesel replied:
OK.  I'll buy that.

jgibson commented on package.json:
```diff
@@ -14,12 +14,14 @@
+    "bunyan": "^1.8.9"
   },
   "devDependencies": {
     "chai": "^3.5.0",
```
bunyan and chai are used like peer dependencies. I think that we haven't hit any errors because new versions of those libraries haven't really been released, however if shr-test-helpers' bunyan or chai became desynced from a parent module we could probably run into issues.

cmoesel replied:
OK.  I buy that argument.  Neither have caused any problems thus far, but it probably makes sense.

jgibson reviewed (changes requested):
`bunyan`, `chai`, and `mocha` are all debatable, but I think that we should do something about `shr-models` 5.2.1 before releasing it.

cmoesel committed:
Dependency changes based on review

Based on code review, made the following changes:
- removed unused mocha dependency
- changed chai to peer dependency
- changed bunyan to peer dependency

cmoesel committed:
Bump shr-models to ^5.2.1

Since these tests rely on a bug fixed in shr-models 5.2.1, require at least that version in the dependency.
